Include Compiled Node Attributes in run_results.json

peterallenwebb opened the pull request:
resolves #7519

### Problem


### Solution


### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

gshank reviewed (commented):
tests/functional/invariants/invariants.py is kind of odd naming. Why "invariants"?  Why the doubled up name? Is there some reason you put "get_run_results" in that file instead of in dbt/tests/util.py along with "get_manifest"? 

Looks like artifacts tests need to be fixed...

peterallenwebb commented:
> tests/functional/invariants/invariants.py is kind of odd naming. Why "invariants"? Why the doubled up name? 

@gshank I agree this was awkwardly named. The conditions I wound up calling "assertions" were initially "invariants" [for nerdy reasons](https://en.wikipedia.org/wiki/Invariant_(mathematics)#Invariants_in_computer_science), but I thought calling them assertions was more friendly, and I should have made the naming consistent.

Let me know what you think of the new names I chose.

> Is there some reason you put "get_run_results" in that file instead of in dbt/tests/util.py along with "get_manifest"?

Only that it didn't occur to me. I've moved the code to tests/util.py.

> Looks like artifacts tests need to be fixed...

Fixed!

gshank reviewed (approved):
Looks good!
